lib/src/http_headers.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+
+import 'globals.dart';
+import 'third_party/cronet/generated_bindings.dart';
+
+/// Headers for HTTP requests.
+///
+/// In some situations, headers are immutable:
```

Comment:
If the other situation does not occur in our implementation. It doesn't make sense to make a bullet point list.

Maybe just have:

```
/// [HttpClientRequest] have immutable headers from the moment the body is
/// written to.
```